docs/content/tutorials/oidc-custom-configuration.md

```diff
@@ -0,0 +1,199 @@
+---
+title: NGINX Ingress Controller and Open Service Mesh
+description: |
+  Use NGINX Ingress Controller with Open Service Mesh.
+weight: 1800
+doctypes: ["concept"]
+toc: true
+docs: "DOCS-1181"
+---
+
+# OIDC Custom Configuration
+
+The F5 NGINX Ingress Controller implements OpenID Connect (OIDC) using the NGINX OpenID Connect Reference implementation: [nginx-openid-connect](https://github.com/nginxinc/nginx-openid-connect).
+
+This guide will walk through how to customise and configure this default implementation.
+
+## Prerequisites
+This guide assumes that you have an F5 NGINX Ingress Controller deployed. If not, please follow the installation steps using either the [Manifest](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/) or [HELM](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-helm/) approach.
+
+To customise the NGINX OpenID Connect Reference implementation, we will need to:
+1. Create a ConfigMap containing the contents of the default `oidc.conf` file
+2. Attach a `Volume` and `VolumeMount` to our deployment of the F5 NGINX Ingress Controller
+
+This setup will allow our custom configuration in our ConfigMap to override the contents of the default `oidc.conf` file.
+
+## Step 1 - Creating the ConfigMap
+
+Run the below command to generate a ConfgMap with the contents of the `oidc.conf` file.
+**NOTE** The ConfgMap must be deployed in the same `namespace` as the F5 NGINX Ingress Controller.
+```
+kubectl create configmap oidc-config-map --from-literal=oidc.conf="$(curl -k https://raw.githubusercontent.com/nginxinc/kubernetes-ingress/v3.1.1/internal/configs/oidc/oidc.conf)"
+```
+
+Use the `kubectl describe` command to confirm the contents of the ConfigMap are correct
+
+```
+kubectl describe configmap oidc-config-map
+```
+
+```
+Name:         oidc-config-map
+Namespace:    default
+Labels:       <none>
+Annotations:  <none>
+
+Data
+====
+oidc.conf:
+----
+    # Advanced configuration START
+    set $internal_error_message "NGINX / OpenID Connect login failure\n";
+    set $pkce_id "";
+    # resolver 8.8.8.8; # For DNS lookup of IdP endpoints;
+    subrequest_output_buffer_size 32k; # To fit a complete tokenset response
+    gunzip on; # Decompress IdP responses if necessary
+    # Advanced configuration END
+
+    ...
+    # Rest of config ammended
+```
+
+## Step 2 - Customising the default configuration
+Once the contents of the `oidc.conf` file has been added to the ConfigMap, you are free to customise the contents of this ConfigMap.
+In this example, we will add a comment to the top of the file to demonstrate it is overwriting the default contents.
```

Comment:
```suggestion
In this example, you can add a comment to the top of the file to demonstrate it is overwriting the default contents.
```

Reply:
How does this sound instead?
```
This example demonstrates adding a comment to the top of the file to demonstrate it is overwriting the default contents.
```

Reply:
This is a great alternative, too - good to go forward with it, though I would change "This comment" to "The comment": cuts down on the repetition of "This", particularly with how concise the sentences are.